master/internal/experiment_filter.go

```diff
@@ -214,7 +214,7 @@ func hpToSQL(c string, filterColumnType *string, filterValue *interface{},
 			queryArgs = append(queryArgs, bun.Safe("?"), queryValue)
 			queryString = fmt.Sprintf(`(CASE
 				WHEN config->'hyperparameters'->%s->>'type' = 'const' THEN config->'hyperparameters'->%s->>'val' NOT LIKE %s
-				WHEN config->'hyperparameters'->%s->>'type' = 'categorical' THEN (config->'hyperparameters'->%s->>'vals')::jsonb %s %s) IS NOT TRUE
+				WHEN config->'hyperparameters'->%s->>'type' = 'categorical' THEN (config->'hyperparameters'->%s->>'vals')::jsonb %s %s IS NOT TRUE
```

Comment:
drive by review since this came up in my slack threads

Can we include tests? This endpoint is really hard to change just because of all the possible dynamic sql that can build. It would be cool if it was exhaustively tested / close to 100% test coverage.

Reply:
i would rather that be in a separate pr -- as you mentioned, there's a lot of possible dynamic sql that can be built. exhaustive testing would add a lot more code to review.

Reply:
Can we include a test for just this regression then?

and make a separate ticket

Reply:
sounds like a good compromise -- will do!

Reply:
I guess this bug is the reason why the current tests are not great. They didn't catch this bug.

Like it's great that we can assert the sql looks some way but it doesn't catch generating wrong sql. Nothing in that test makes me think this code is actually fixing the issue.

What I think would be a lot better test would be to actually run a query with Not Contains against the database and make sure that it doesn't error and filters as expected.

Reply:
gotcha -- added!